banzai_nres/dbs.py

```diff
@@ -0,0 +1,64 @@
+from banzai.dbs import Base, create_db
+from sqlalchemy import Column, String, Integer, Float, Index
+import boto3
+import banzai.dbs
+import os
+from glob import glob
+
+
+class PhoenixModel(Base):
+    __tablename__ = 'phoenixmodels'
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    filename = Column(String(100), unique=True)
+    location = Column(String(150))
+    T_effective = Column(Float)
+    log_g = Column(Float)
+    metallicity = Column(Float)
+    alpha = Column(Float)
+    Index('idx_model', 'T_effective', 'log_g', 'metallicity', 'alpha')
+
+
+def create_nres_db(db_address):
+    create_db(db_address)
+
+
+def populate_phoenix_models(model_location, db_address):
+    if 's3' in model_location:
+        s3 = boto3.resource('s3')
+        model_bucket = s3.Bucket(model_location.replace('s3://', ''))
+        model_files = model_bucket.objects.all()
+    else:
+        # Assume they are on disk
+        model_files = glob(os.path.join(model_location, '*.fits'))
+    with banzai.dbs.get_session(db_address) as db_session:
+        # strip off the s3
+        for model_file in model_files:
+            if 's3' in model_location:
+                filename = model_file.key
+                location = model_location
+            else:
+                filename = os.path.basename(model_file)
+                location = os.path.dirname(model_file)
+
+            if 'wave' in filename.lower():
+                continue
+            equivalence_criteria = {'filename': filename}
+            _, T_effective, log_g, metallicity, alpha = os.path.splitext(filename)[0].split('_')
```

Comment:
This will work but it feels fragile. I think I understand why you're doing it, since you only have access to the filenames and not the contents themselves?

Any way to define this naming convention somewhere as a setting/global? Or at least have this convention explicitly stated somewhere.

Another way of making this work would be to have an index file in the bucket with every catalog entry and it's corresponding metadata (Teff, logg, metallicity, etc...). Not totally necessary and may be overkill, but I'll leave that up to you.

Reply:
You aren't wrong. In the munge_phoenix_models we define the filename convention.